ooni/reporter.py

```diff
@@ -298,8 +299,13 @@ def createReport(self):
             self.agent = SOCKS5Agent(reactor, proxyEndpoint=proxyEndpoint)
 
         elif self.collectorAddress.startswith('https://'):
-            # XXX add support for securely reporting to HTTPS collectors.
-            log.err("HTTPS based collectors are currently not supported.")
+            # not sure if there's something else it needs.  Seems to work.
+            # Very difficult to get it to work with self-signed certs.
+            self.agent = Agent(reactor)
```

Comment:
@hellais I don't know enough about Twisted's Agent and certificates to judge whether this can be merged or further work is needed. From my understanding of #395 review, #401 adds Twisted 15.0 support and Twisted 15.0 is the minimum required to guarantee that certificates are securely accepted. Shall we also raise the minimum Twisted to 15.0 because of that, or otherwise how do you suggest to proceed here?